lib/compiler.js

```diff
@@ -75,11 +74,25 @@ exports.runWebpackCompiler = function runWebpackCompiler(config, $mobileHelper,
                 cwd: $projectData.projectDir
             });
 
+            let isFirstWebpackWatchCompilation = true;
             function resolveOnWebpackCompilationComplete(message) {
                 if (message === messages.compilationComplete) {
-                    console.log("Initial webpack build done!");
+                    console.log("Webpack build done!");
                     resolve();
                 }
+
+                if (message.emittedFiles) {
+                    if (isFirstWebpackWatchCompilation) {
+                        isFirstWebpackWatchCompilation = false;
+                        return;
+                    }
+
+                    if (hookArgs.filesToSync && hookArgs.startSyncFilesTimeout) {
+                        // TODO: Might need to get the "app" constant from config file in the future
```

Comment:
Let's remove the 'TODO's and create a separate issue, tracking all the changes that may be needed in the plugin after https://github.com/NativeScript/android-runtime/pull/944 is merged.